org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/preferences/BuilderPreferenceAccess.java

```diff
@@ -32,6 +33,37 @@
 	 * Name of a preference for configuring whether the builder participant is enabled or not.
 	 */
 	public static final String PREF_AUTO_BUILDING = "autobuilding"; //$NON-NLS-1$
+	/**
+	 * Name of a preference defining the scheduling rule strategy for the builder.
+	 * @since 2.16
+	 */
+	@Beta
+	public static final String PREF_SCHEDULING_RULE = "schedulingrule"; //$NON-NLS-1$
+	/**
+	 * Options for scheduling rules.
+	 * @since 2.16
+	 */
+	public static enum SchedulingOption {
+		WORKSPACE ("workspace"),
+		// all Xtext projects in the workspace
+		XTEXT_PROJECTS ("all_xtext_projects"), //$NON-NLS-1$
+		// current built project
+		PROJECT ("project");
+		private String name;
```

Comment:
Enums do already have a name. We should not overload that.

Reply:
Agreed. I wanted to have the values lower case and the enum literal upper case. But it is OK to have the value upper case I guess.